experimental/PlaneCurve/ProjEllipticCurve.jl

```diff
@@ -0,0 +1,460 @@
+export ProjEllipticCurve, discriminant, issmooth, j_invariant,
+       Point_EllCurve, curve, weierstrass_form, toweierstrass,
+       iselliptic
+
+################################################################################
+# Helping functions
+################################################################################
+
+function isweierstrassform(eq::Oscar.MPolyElem{S}) where S <: FieldElem
+   R = parent(eq)
+   x = gen(R, 1)
+   y = gen(R, 2)
+   z = gen(R, 3)
+   A = [x*y*z, y*z^2, x^2*z, x*z^2, z^3]
+   f = deepcopy(eq)
+   f = f - y^2*z + x^3
```

Comment:
the arithmetic operations should not be modifing their inputs - why not just have
```julia
f = eq - y^2*z + x^3
```
here?

Reply:
You are right, thank you for the comment !